### PR TITLE
feat: `hoistTry` rule

### PR DESCRIPTION
**Pull request by facumenzella**
## What? 
Adding `hoistTry` rule based on [this](https://github.com/nicklockwood/SwiftFormat/issues/948) issue. 

The PR consists of commits:
1. Basic rule implementation
2. Added postfix related code to avoid hoisting those.
3. Added a few more tests.

### CODE REVIEW

**Comment by nicklockwood** [comment]
Amazing, thank you! I was just thinking about adding a hoistTry rule based on your hoistAwait, but I was concerned about all the edge cases, such as `try?` and ensuring `try await` order isn't changed. I see you've covered at least some of these in your tests though.

File: Sources/Rules.swift [hunk 1]
```diff
@@ -4292,6 +4292,91 @@ public struct _FormatRules {
...
+            tryIndexes.reversed()
+                .forEach { tryIndex in
+                    formatter.removeToken(at: tryIndex)
+
+                    if formatter.tokens[tryIndex].isSpace == true {
+                        formatter.removeToken(at: tryIndex)
+                    }
+                }
+
+            if let firstSpaceOrLinebreakIndex = formatter.index(before: idx, where: { $0.isSpaceOrLinebreak }),
```
Comment (nicklockwood) [inline Sources/Rules.swift:4364]: For future reference you can use `formatter.index(of: .spaceOrLinebreak, before: idx)` in cases like this.
